src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1268,32 +1268,56 @@ public Block FindBlock(Keccak blockHash, BlockTreeLookupOptions options)
 
         private void SetTotalDifficulty(BlockHeader header)
         {
-            if (header.TotalDifficulty != null)
+            BlockHeader GetParentHeader(BlockHeader current) => 
+                this.FindParentHeader(current, BlockTreeLookupOptions.TotalDifficultyNotNeeded) 
```

Comment:
TotalDifficultyNotNeeded?

Reply:
yes, exactly

Without this flag FindParentHeader will calculate total difficulty and if its not present it results in recursion. We want to avoid recursion and have a  separate code here that will do exactly that - calculate total difficulty, but iterativly.